packages/seal/src/key-server.ts

```diff
@@ -62,29 +58,40 @@ export async function retrieveKeyServers({
 	objectIds: string[];
 	client: SealCompatibleClient;
 }): Promise<KeyServer[]> {
-	// todo: do not fetch the same object ID if this is fetched before.
 	return await Promise.all(
 		objectIds.map(async (objectId) => {
-			let res;
-			try {
-				res = await client.core.getObject({
-					objectId,
-				});
-			} catch (e) {
-				throw new InvalidGetObjectError(`KeyServer ${objectId} not found; ${(e as Error).message}`);
-			}
-
+			// First get the KeyServer object and validate it.
+			const res = await client.core.getObject({
+				objectId,
+			});
 			const ks = KeyServerMove.parse(res.object.content);
-			if (ks.keyType !== 0) {
-				throw new UnsupportedFeatureError(`Unsupported key type ${ks.keyType}`);
+
+			if (
+				Number(ks.lastVersion) > EXPECTED_SERVER_VERSION ||
```

Comment:
hmm, this condition is wrong, no?
should be if (last < expected || first > expected)

Reply:
let's add a test to catch those

Reply:
we agreed in the past that first_version = latest version. e.g. 2 (last), 3, 4, 5 (first) so if expected > 5 or expected < 2 throws error. 

we can still flip it though if you would like to think first_version = oldest

Reply:
oh, i didn't remember that, it's a bit confusing that last<=first, no? do we depend on that elsewhere?

Reply:
yes i agreed to that! we can flip it here if we want to. it's not used anywhere since we only initialized with 1 for both values.

Reply:
let's switch